src/e2e/app/cli/common/src/kamu_api_server_client_ext.rs

```diff
@@ -1020,36 +1259,130 @@ pub struct DatasetDependencies {
     pub downstream: Vec<DatasetDependency>,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AccountWithRole {
+    pub account_id: odf::AccountID,
+    pub account_name: odf::AccountName,
+    pub role: AccountToDatasetRelation,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AccountRolesResponse {
+    pub accounts_with_roles: Vec<AccountWithRole>,
+}
+
+impl AccountRolesResponse {
+    pub fn new(accounts_with_roles: Vec<AccountWithRole>) -> Self {
+        Self {
+            accounts_with_roles,
+        }
+    }
+}
+
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 #[derive(Error, Debug)]
 pub enum GetDatasetVisibilityError {
-    #[error("Forbidden")]
-    Forbidden,
-    #[error("Not found")]
-    NotFound,
+    #[error("Dataset not found")]
+    DatasetNotFound,
     #[error(transparent)]
     Internal(#[from] InternalError),
 }
 
+impl PartialEq for GetDatasetVisibilityError {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::DatasetNotFound, Self::DatasetNotFound) => true,
+            (Self::Internal(a), Self::Internal(b)) => a.reason().eq(&b.reason()),
+            (_, _) => false,
+        }
+    }
+}
+
 #[derive(Error, Debug)]
 pub enum SetDatasetVisibilityError {
-    #[error("Not found")]
-    NotFound,
+    #[error("Dataset not found")]
+    DatasetNotFound,
+    #[error("Access")]
+    Access,
     #[error(transparent)]
     Internal(#[from] InternalError),
 }
 
+impl PartialEq for SetDatasetVisibilityError {
```

Comment:
1) Normally the errors are not compared, rather tested with `assert_matches!`

2) If you customize `PartialEq`, is there a reason not to define `Eq`?

Reply:
Done